test.js

```diff
@@ -89,6 +91,65 @@ test('metrics', async function (t) {
   t.alike(swarm1.stats.commands.ping, { tx: 1, rx: 1 }, 'ping resp')
 })
 
+test('delayed ping', async function (t) {
+  const [swarm1, swarm2] = await makeSwarm(2, t)
+  const start = Date.now()
+  await swarm1.delayedPing({ host: swarm2.host, port: swarm2.port }, 1_100)
+  const end = Date.now()
+  t.ok(end - start >= 1_100, 'ping delayed for at least 1.1 seconds')
+})
+
+test('delayed ping - rejects when delay exceeds client cap', async function (t) {
+  const [swarm1, swarm2] = await makeSwarm(2, t)
+  t.exception(
+    async () => await swarm1.delayedPing({ host: swarm2.host, port: swarm2.port }, 10_001),
+    /Delay exceeds max delay: 10000ms/
+  )
+})
+
+test('delayed ping - server times out when exceeding delay via internal request', async function (t) {
```

Comment:
Optional: now that `maxDelayMs` is a constructor opt you can re-write the test to use the normal API (setting a bigger value for the client than for the server)